core/selection.cpp

```diff
@@ -231,18 +222,47 @@ bool setSelection(const std::vector<dive *> &selection, dive *currentDive, int c
 	if (currentDc >= 0)
 		dc_number = currentDc;
 	fixup_current_dc();
+	return divesToSelect;
+}
 
-	// Send the new selection
-	emit diveListNotifier.divesSelected(divesToSelect);
-	return current_dive != oldCurrent;
+static void clear_trip_selection()
+{
+	amount_trips_selected = 0;
+	for (int i = 0; i < trip_table.nr; ++i)
+		trip_table.trips[i]->selected = false;
 }
 
-bool setSelection(const std::vector<dive *> &selection)
+// Reset the selection to the dives of the "selection" vector and send the appropriate signals.
+// Set the current dive to "currentDive". "currentDive" must be an element of "selection" (or
+// null if "selection" is empty).
+void setSelection(const std::vector<dive *> &selection, dive *currentDive, int currentDc)
+{
+	// Since we select only dives, there are no selected trips!
+	clear_trip_selection();
+
+	auto selectedDives = setSelectionCore(selection, currentDive, currentDc);
+
+	// Send the new selection to the UI.
+	emit diveListNotifier.divesSelected(selectedDives);
```

Comment:
stylistically you could have just done one nested call and avoided the auto variable - but the compiler will do this anyway, so... meh

Reply:
True. Personally I find nested calls harder to read.

Reply:
I go back and forth... I think here the line would get pretty long.